plugins/inputs/opcua_listener/subscribe_client.go

```diff
@@ -0,0 +1,151 @@
+package opcua_listener
+
+import (
+	"context"
+	"fmt"
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/ua"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/opcua/input"
+	"reflect"
+	"time"
+)
+
+type SubscribeClientConfig struct {
+	input.InputClientConfig
+	SubscriptionInterval config.Duration `toml:"subscription_interval"`
+}
+
+type SubscribeClient struct {
+	*input.OpcUAInputClient
+	Config SubscribeClientConfig
+
+	sub                *opcua.Subscription
+	monitoredItemsReqs []*ua.MonitoredItemCreateRequest
+	dataNotifications  chan *opcua.PublishNotificationData
+	metrics            chan telegraf.Metric
+
+	processingCtx    context.Context
+	processingCancel context.CancelFunc
+}
+
+func (sc *SubscribeClientConfig) CreateSubscribeClient(log telegraf.Logger) (*SubscribeClient, error) {
+	client, err := sc.InputClientConfig.CreateInputClient(log)
+	if err != nil {
+		return nil, err
+	}
+
+	subClient := &SubscribeClient{
+		OpcUAInputClient:   client,
+		Config:             *sc,
+		monitoredItemsReqs: make([]*ua.MonitoredItemCreateRequest, len(client.NodeIDs)),
+		dataNotifications:  make(chan *opcua.PublishNotificationData, 100), // TODO: Make this a setting or a calculation, maybe?,
+		metrics:            make(chan telegraf.Metric, 100),                // TODO: Make this a setting or a calculation, maybe?,
```

Comment:
What would a user set these two? I'd like to see ideas and remove the TODO if possible

Reply:
I'm not entirely sure. It should be big enough to prevent the  channels from blocking. A notification is received when the subscription on the opcua monitored item reports a change. The notification is then used to create a metric. It might be good enough to set the size to the number of nodes that are subscribed to, but this might be cutting it close if they all reports changes at the same time. Maybe a multiple of the number of nodes might work. What do you think?

I think making the user choose something will be a bad idea, as it is an implementation detail.

Reply:
> making the user choose something will be a bad idea,

Agreed

Have you been using this plugin and found your defaults are working for now?

Reply:
The defaults are working for us, however. We do not currently have a lot of data going through this system. At most 30 items, of which at most 8 at10 Hz. Others only once every few minutes.

Reply:
Ah I missed this comment. I'd remove the TODO and maybe leave a comment above both of these as to why these were picked.

Reply:
Alright, added an explanation and a possible alternative if these values prove insufficient.